corehq/apps/sso/models.py

```diff
@@ -279,6 +279,20 @@ def does_domain_trust_user(cls, domain, username):
         return idp.does_domain_trust_this_idp(domain)
 
 
+@receiver(pre_save, sender=IdentityProvider)
```

Comment:
Same argument against signals here. This logic could be handled by overriding save with a `clear_cache` boolean -- although I think its simpler to just clear the cache on every save.

Reply:
moved to `save()`